collect_app/src/main/java/org/odk/collect/android/preferences/PreferencesActivity.java

```diff
@@ -62,6 +64,8 @@ public void onBackPressed() {
         } else {
             super.onBackPressed();
         }
+        Intent intent = new Intent();
```

Comment:
It doesn't right to me to always set a specific result when leaving the `PreferencesActivity`. This could happen in the `UserInterfacePreferences` fragment when the font size itself changes. You can call set result on a Fragment's host Activity by calling `getActivity()`. 

It would also be nice to write a test for the Fragement (probably using Robolectric) to check that it sets the result on it's parent Activity.

Reply:
Ah sorry ignore me. Looking at the issue again (#3356) it seems like we want to refresh the form when any setting changes so this actually does feel right.

Reply:
Can you @seadowg sir, please guide what do from here.